go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -112,6 +109,33 @@ func transformFkCascade(ctx *plancontext.PlanningContext, fkc *operators.FkCasca
 	return newFkCascade(parentLP, selLP, children), nil
 }
 
+func transformSubQueryFilter(ctx *plancontext.PlanningContext, op *operators.SubQuery) (logicalPlan, error) {
+	outer, err := transformToLogicalPlan(ctx, op.Outer)
+	if err != nil {
+		return nil, err
+	}
+
+	inner, err := transformToLogicalPlan(ctx, op.Subquery)
+	if err != nil {
+		return nil, err
+	}
+
+	cols, err := op.GetJoinColumns(ctx, op.Outer)
```

Comment:
Here `GetJoinColumns` is a method, so it doesn't need op.Outer as an input, the `op` already has it.

Reply:
I think this is by design. this method is used from a few different places, and it's not always it's own outer that is used